src/experiments/logins/api.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/* globals AppConstants, browser, Components, CustomizableUI, dispatcher,
+   ExtensionCommon, ExtensionAPI, Services, XPCOMUtils */
+
+"use strict";
+
+ChromeUtils.defineModuleGetter(this, "ExtensionUtils",
+                               "resource://gre/modules/ExtensionUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "LoginHelper",
+                               "resource://gre/modules/LoginHelper.jsm");
+ChromeUtils.defineModuleGetter(this, "Services",
+                               "resource://gre/modules/Services.jsm");
+
+const { ExtensionError } = ExtensionUtils;
+
+const getLogins = () => {
```

Comment:
The existing APIs seem to move functions out to API-global scope, rather than have helper functions defined on the ExtensionAPI subclass, so I went with that approach for `getLogins` and `getLogin`.